### DISCUSSION TITLE
'Copy Image' in Firefox not working

### DISCUSSION THREAD

**Discussion by laur89** [hluk/CopyQ#2066] [Q&A] [answered]
If an image is right-clicked in FF, and `Copy Image` option is selected, then nothing is shown in copyq clipboard.
On the other hand, if `Copy image` is selected in Chrome or Chromium, then an item appears and it can be pasted into applications that support images, eg Signal and Slack. Is this a bug or a user/configuration error?

Separate question, but still related: after image is copied in Chrome, it shows in copyq clipboard as text `https://...` link (inspecting via `F4` also lists only `text/plain`), but if it's immediately pasted into say Slack it works as expected - an image is pasted.
However, if we explicitly select this item from copyq clipboard after copying, then it starts acting as text - when pasted into same Slack window, it now shows up as text link as opposed to an image. What's up with that?

```
 copyq --version
CopyQ Clipboard Manager 6.2.0
Qt: 5.15.4
KNotifications: 5.94.0
Compiler: GCC
Arch: x86_64-little_endian-lp64
OS: Debian GNU/Linux bookworm/sid
```
FF `104.0`

**Comment by hluk** [comment, marked as answer]
I just noticed a bug in "Ignore items with no or single character" predefined command (fix in #2074) - it would not store items with any text but also image items. Are you by any chance using it? (It would be listed in Commands dialog - F6 shortcut.)

> after image is copied in Chrome, it shows in copyq clipboard as text

Usually only `text/html` and `image/png` is stored. I tested this with copying avatar from this discussion page and CopyQ stores it without `text/plain` when copied from Firefox and Chromium. Maybe another command in CopyQ is adding the text?

Do you have Images enabled in Preference, Items tab?
![image](https://user-images.githubusercontent.com/38054/188299956-4b9e65c6-eb81-4f73-8a28-5f2d39c9b59d.png)

**Reply by laur89** [reply]
`Preference->Items` contains no items whatsoever:
![image](https://user-images.githubusercontent.com/4551018/188303517-63bc2845-68c6-42fa-b6ff-7036d616d208.png)

I'm sure it works though, as it works as expected when explicitly copying images via scripting:
```
copyq write "$file_mime" - < "$input_file" && copyq select 0
```

Removing all commands & restarting copyq made no difference to image copy behavior.

**Reply by hluk** [reply]
Looks like plugins are not installed. On some Linux distributions there is **a separate package for plugins** (`copyq-plugins`). Can you try installing it and checking again?

**Reply by laur89** [reply]
God damnit, all these years wasted! Thanks for pointing that package out, did the trick.

**Reply by laur89** [reply]
Edit: on another note, after installing the plugins, the `copyq-commands.ini` `size` property value went from 5 to 12, yet it added only 4 extra commands (meaning i'd expect the value to be increased to 9 instead); is this an issue?
- [Pre-plugins package commands.ini](https://github.com/laur89/dotfiles/blob/852c4fda46cd766f677c00b226701fa02ab96fdf/home/.config/copyq/copyq-commands.ini)
- [Post-plugins commands.ini](https://github.com/laur89/dotfiles/blob/f5abd6a39baabbd737d81bbb5ed8b4db0f95018e/home/.config/copyq/copyq-commands.ini)

(note the auto-update of the file collated the linebreaks of the commands, which I always manually revert to keep the raw file more readable)

**Reply by hluk** [reply]
There are some commands that the plugins add automatically. If you do not need them, disable the plugins or the command. Note: If you just remove the commands without disabling the plugins they will be added again. This is a still quite messy, I would like to improve it somehow, hopefully in the near future.
